### ISSUE TITLE
Check for side effects of self test recommended

### ISSUE THREAD

**Issue by HolQue** [test-fullautomation/robotframework-testsuitesmanagement#128]
The self test of RobotFramework_TestsuitesManagement (`robotframework-testsuitesmanagement\atest`) requires to call the Robot Framework with several command lines. Because the RobotFramework_TestsuitesManagement is a component of the Robot Framework and because of the self test already starts at command line level, I assume, that also the Robot Framework itself is a part of the "system under test" (and not the RobotFramework_TestsuitesManagement only).

Therefore the Robot Framework itself should not be used to execute the self tests. Another process, e.g. a separate Python script, has to do this. Otherwise we cannot be sure that both involved processes (the one who executes the test and the one who is tested) are strongly separated from each other.

This is in opposite to the current implementation: Here a robot file together with the keyword `Subprocess Execution` is used to execute the self test.

I recommend to check if this really has no unwanted side effects - but maybe as a basic principle it's better to keep both processes separated from each other.

**Comment by namsonx** [comment]
Hello Holger,

First of all, the atest test cases of RobotFramework_Testsuitesmanagement were failed in Linux due to my mistake, I forgot maintenance the atest_libs.py for OSD operating system.
I had created the pull-request #131 to update the atest_libs for OSD.

I had been looking in some documents about subprocess in Python, it seems there is no side effect or major difference between Windows and Linux.  

Back to the atest of RobotFramework_TestsuitesManagement, I defined many test cases to test the Robot configuration features which we offered in RobotFramework_TestsuitesManagement. To avoid the mutual influence between the tests regarding to robot configuration object and robot global variables, I have to separate these test cases into distinguishable processes.

For the robot test scripts which used to call subprocess to execute another robot script containing RobotFramework_TestsuitesManagement just normal robot test script which verify the console log of robot script under test.

Thank you,
Son
